### PR TITLE
HPCC-25380 Add a class for catching incorrect use of IFileIO

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
The main issue it caught was the code to read the environment was using a 32K buffer.  There are also various issues in ftslave which I have not addressed.  I suspect I should increase the checking sizes to 1Mb - but that is a separate discussion.

File: system/jlib/jfile.ipp [hunk 1]
```diff
@@ -146,6 +142,38 @@ protected:
...
+    void report(const char * format, ...) __attribute__((format(printf, 2, 3)));
+
+protected:
+    CriticalSection cs;
+    StringAttr filename;
+    Linked<IFileIO> io;
+    bool closed = false;
+    bool traced = false;
+    unsigned minSeqReadSize = 0x10000;
+    unsigned minWriteSize = 0x010000;
```
Comment (jakesmith) [inline system/jlib/jfile.ipp:172]: might want to make params to class, or helper to set them.
So when/if we have different defaults, depending on FS type, they can be configured.

(probably can wait for when we actually need it).

**Review by jakesmith — APPROVED** [review]
@ghalliday - looks good.

**Comment by AttilaVamos** [comment]
This PR went timeout in Smoketest.

**Pull request merged**